bundles/org.openhab.binding.verisure/src/main/java/org/openhab/binding/verisure/internal/discovery/VerisureThingDiscoveryService.java

```diff
@@ -0,0 +1,177 @@
+        implements DiscoveryService, ThingHandlerService {
+
+    private static final int SEARCH_TIME_SECONDS = 60;
+    private final Logger logger = LoggerFactory.getLogger(VerisureThingDiscoveryService.class);
+
+    private @NonNullByDefault({}) VerisureBridgeHandler verisureBridgeHandler;
+    private @NonNullByDefault({}) ThingUID bridgeUID;
+
+    public VerisureThingDiscoveryService() {
+        super(VerisureHandlerFactory.SUPPORTED_THING_TYPES, SEARCH_TIME_SECONDS);
+    }
+
+    @Override
+    public void startScan() {
+        removeOlderResults(getTimestampOfLastScan());
+        logger.debug("VerisureThingDiscoveryService:startScan");
+        if (verisureBridgeHandler != null) {
+            VerisureSession session = verisureBridgeHandler.getSession();
+            if (session != null) {
+                HashMap<String, VerisureThing> verisureThings = session.getVerisureThings();
+                for (Map.Entry<String, VerisureThing> entry : verisureThings.entrySet()) {
+                    VerisureThing thing = entry.getValue();
+                    if (thing != null) {
+                        logger.debug("Thing: {}", thing);
+                        onThingAddedInternal(thing);
+                    }
+                }
+            }
+        }
+    }
+
+    private void onThingAddedInternal(VerisureThing thing) {
+        logger.debug("VerisureThingDiscoveryService:OnThingAddedInternal");
+        ThingUID thingUID = getThingUID(thing);
+        String deviceId = thing.getDeviceId();
+        if (thingUID != null && deviceId != null) {
+            if (verisureBridgeHandler != null) {
+                String label = "Device Id: " + deviceId;
+                if (thing.getLocation() != null) {
+                    label += ", Location: " + thing.getLocation();
+                }
+                if (thing.getSiteName() != null) {
+                    label += ", Site name: " + thing.getSiteName();
+                }
+                DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withBridge(bridgeUID)
+                        .withLabel(label).withProperty(VerisureThingConfiguration.DEVICE_ID_LABEL, deviceId).build();
+                logger.debug("thinguid: {}, bridge {}, label {}", thingUID.toString(), bridgeUID, thing.getDeviceId());
+                thingDiscovered(discoveryResult);
+            }
+        } else {
+            logger.debug("Discovered unsupported thing of type '{}' with deviceId {}", thing.getClass(),
+                    thing.getDeviceId());
+        }
+
+    }
+
+    private @Nullable ThingUID getThingUID(VerisureThing thing) {
+        ThingUID thingUID = null;
+        if (verisureBridgeHandler != null) {
+            String deviceId = thing.getDeviceId();
+            if (deviceId != null) {
+                // Make sure device id is normalized, i.e. replace all non character/digits with empty string
+                deviceId.replaceAll("[^a-zA-Z0-9]+", "");
```

Comment:
Strings are immutable. `replaceAll` returns a new string that is getting thrown away here.

Reply:
Oops, nice finding! Fixed!